framework/src/materials/CoupledGradientMaterial.C

```diff
@@ -0,0 +1,48 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "CoupledGradientMaterial.h"
+
+registerMooseObject("MooseApp", CoupledGradientMaterial);
+registerMooseObject("MooseApp", ADCoupledGradientMaterial);
+
+template <bool is_ad>
+InputParameters
+CoupledGradientMaterialTempl<is_ad>::validParams()
+{
+  InputParameters params = Material::validParams();
+  params.addClassDescription("Create a gradient material coupled to a variable.");
+  params.addRequiredParam<MaterialPropertyName>(
+      "grad_mat_prop", "Name of the gradient material property that will be created.");
```

Comment:
how can we reword to make it clear this is not a graidnet of a material property

Reply:
"Name of gradient material property equal to the gradient of the coupled gradient times the scalar"?

Reply:
gradient of the coupled variable?

Reply:
oh, yeah. coupled variable gradient